File: file_manipulation/csv_files/csv_example.java
import org.apache.spark.sql.*;
import org.apache.spark.sql.types.*;
import java.util.Arrays;
import java.util.List;

StructType schema = new StructType(new StructField[]{
    DataTypes.createStructField("id", DataTypes.IntegerType, false),
    DataTypes.createStructField("nombre", DataTypes.StringType, true),
    DataTypes.createStructField("ciudad", DataTypes.StringType, true)
});

List<Row> data = Arrays.asList(
    RowFactory.create(1, "Carolina", "Medellín"),
    RowFactory.create(2, null, "Islandia")
);

Dataset<Row> df = spark.createDataFrame(data, schema);

df.write().mode(SaveMode.Overwrite)
    .option("header", "true")
    .option("nullValue", "null")
    .csv("path_java");

File: file_manipulation/csv_files/csv_example.scala
import org.apache.spark.sql.types._
import org.apache.spark.sql.SaveMode

val schema = StructType(List(
  StructField("id", IntegerType, false),
  StructField("nombre", StringType, true),
  StructField("ciudad", StringType, true)
))

val data = Seq(Row(1, "Caro", "Medellín"), Row(2, null, "Bogotá"))
val df = spark.createDataFrame(spark.sparkContext.parallelize(data), schema)

// Escritura optimizada
df.write
  .option("header", "true")
  .option("charset", "UTF-8")
  .option("nullValue", "NA")
  .mode(SaveMode.Overwrite)
  .csv("output_scala")